src/newtondynamics_physics/SysNewton.h

```diff
@@ -52,12 +50,30 @@ class SysNewton
 
     using ActiveEnt                 = osp::active::ActiveEnt;
     using ACtxPhysics               = osp::active::ACtxPhysics;
-    using ACompHierarchy            = osp::active::ACompHierarchy;
+    using ACtxSceneGraph            = osp::active::ACtxSceneGraph;
     using ACompTransform            = osp::active::ACompTransform;
     using ACompTransformControlled  = osp::active::ACompTransformControlled;
     using ACompTransformMutable     = osp::active::ACompTransformMutable;
 public:
 
+    static void cb_force_torque(const NewtonBody* pBody, dFloat timestep, int threadIndex);
+
+    static void cb_set_transform(NewtonBody const* const pBody, const dFloat* const pMatrix, int threadIndex);
+
+    static void resize_body_data(ACtxNwtWorld& rCtxWorld);
+
+    [[nodiscard]] static NewtonCollision* create_primative(
```

Comment:
This lacks lifetime management.

I recommend either returning an std::unique_ptr<>, to clearly signal that the returned pointer isn't managed by the library

or something related to these attributes: https://clang.llvm.org/docs/AttributeReference.html#handle-attributes